src/client/event_handler.rs

```diff
@@ -59,11 +59,17 @@ macro_rules! event_handler {
             }
 
             /// Runs the given [`EventHandler`]'s code for this event.
-            pub async fn dispatch(self, handler: &dyn EventHandler) {
+            ///
+            /// # Panics
+            /// Panics if the event requires [`Context`] and `ctx` is None.
+            pub async fn dispatch(self, handler: &dyn EventHandler, ctx: Option<Context>) {
                 match self { $(
                     $( #[cfg(feature = $feature)] )?
                     Self::$variant_name { $( $arg_name ),* } => {
-                        handler.$method_name( $( $arg_name ),* ).await;
+                        $(let Some($context) = ctx else {
+                            panic!("Missing Context for dispatch call!");
+                        };)?
+                        handler.$method_name($($context,)? $( $arg_name ),* ).await;
                     }
                 )* }
             }
```

Comment:
IMO a better impl is the following: 
```rust
pub async fn dispatch(self, ctx: Context, handler: &dyn EventHandler) {
    match self { $(
        $( #[cfg(feature = $feature)] )?
        Self::$variant_name { $( $arg_name ),* } => {
            $( let $context = ctx; )?
            handler.$method_name($($context,)? $( $arg_name ),* ).await;
        }
    )* }
}
```

Right now, only `FullEvent::Ratelimit` doesn't contain a `ctx`, and it's never dispatched normally. So, making `ctx` optional here doesn't fit as it's basically required anyway.

Reply:
Done